services/ingestion/etl/just/csv/src/lib/transform.js

```diff
@@ -0,0 +1,166 @@
+ * - `Project country`
+ *
+ * @memberof AgriCsvTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {Array} List of {Location} objects for `project_locations` field
+ */
+const getLocations = record => {
+  const projectLongitude = record.field_prj_longitude || '';
+  const projectLatitude = record.field_prj_latitude || '';
+
+  const country = record.field_prj_country_iso;
+  const hasCoordinates = projectLongitude && projectLatitude;
+
+  return [
+    {
+      country_code: (country && getCountryCode(country)) || '',
+      region: '',
+      nuts: [],
+      address: '',
+      postal_code: '',
+      town: '',
+      centroid: hasCoordinates
+        ? {
+            lon: parseFloat(projectLongitude) || 0,
+            lat: parseFloat(projectLatitude) || 0,
+          }
+        : null,
+      location: hasCoordinates
+        ? {
+            type: 'Point',
+            coordinates: [
+              parseFloat(projectLongitude) || 0,
+              parseFloat(projectLatitude) || 0,
+            ],
+          }
+        : null,
+    },
+  ];
+};
+
+/**
+ * Preprocess related links
+ *
+ * Depends on record['Related links'] field
+ *
+ * @memberof AgriCsvTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {Array|Object} List of {RelatedLink}
+ *
+ * @example
+ * input => "<a href=\"https://ec.europa.eu/inea/en/ten-t/ten-t-projects/projects-by-country/multi-country/2013-eu-92069-s\">INEA</a>;<a href=\"https://europa.eu/investeu/projects/central-european-green-corridors_en\">InvestEU</a>"
+ * output => [
+ *    { label: "INEA", url: "https://ec.europa.eu/inea/en/ten-t/ten-t-projects/projects-by-country/multi-country/2013-eu-92069-s" }
+ *    { label: "InvestEU", url: "https://europa.eu/investeu/projects/central-european-green-corridors_en" }
+ *  ]
+ */
+const getRelatedLinks = record =>
+  (record.field_prj_link || '')
+    .split(';')
+    .filter(link => link)
+    .map(link => {
+      const matches = link.match(/<a .*href="(.*)".*>(.*)<\/a>/i);
+
+      if (Array.isArray(matches) && matches.length === 3) {
+        return {
+          url: matches[1],
+          label: matches[2],
+        };
+      }
+
+      return {
+        url: '',
+        label: '',
+      };
+    })
+    .filter(link => link !== null);
+
+/**
+ * Map fields for JUST producer, CSV file types
+ *
+ * Example input data: {@link https://github.com/ec-europa/eubfr-data-lake/blob/master/services/ingestion/etl/just/csv/test/stubs/record.json|stub}
+ *
+ * Transform function: {@link https://github.com/ec-europa/eubfr-data-lake/blob/master/services/ingestion/etl/just/csv/src/lib/transform.js|implementation details}
+ * @name JustCsvTransform
+ * @param {Object} record Piece of data to transform before going to harmonized storage.
+ * @returns {Project} JSON matching the type fields.
+ */
+export default (record: Object): Project => {
+  // Preprocess budget
+  const budgetObject = {
```

Comment:
Could be extracted as a function as the others and hoisted out of the scope of this object return, like the rest.